File: testerum-backend/testerum-runner/testerum-runner-cmdline/src/main/assembly/report_templates/common/testerum-model/model/report/report-step.ts
import {ReportStepCall} from "../step/call/report-step-call";
import {ExecutionStatus} from "./execution-status";
import {ExceptionDetail} from "../exception/exception-detail";
import {MarshallingUtils} from "../../json-marshalling/marshalling-utils";
import {RunnerReportNode} from "./runner-report-node";

export class ReportStep implements RunnerReportNode {

    constructor(public readonly stepCall: ReportStepCall,
                public readonly startTime: Date,
                public readonly endTime: Date,
                public readonly durationMillis: number,
                public readonly status: ExecutionStatus,
                public readonly exceptionDetail: ExceptionDetail | null,
                public readonly textLogFilePath: string,
                public readonly modelLogFilePath: string,
                public readonly children: Array<ReportStep>) {}

    static parse(input: Object): ReportStep {
        const stepCall = ReportStepCall.parse(input["stepCall"]);
        const startTime = MarshallingUtils.parseLocalDateTime(input["startTime"]);
        const endTime = MarshallingUtils.parseLocalDateTime(input["endTime"]);
        const durationMillis = input["durationMillis"];
        const status = MarshallingUtils.parseEnum(input["status"], ExecutionStatus);
        const exceptionDetail = ExceptionDetail.parse(input["exceptionDetail"]);
        const textLogFilePath = input["textLogFilePath"];
        const modelLogFilePath = input["textLogFilePath"];
        const children = MarshallingUtils.parseList(input["children"], ReportStep);

        return new ReportStep(stepCall, startTime, endTime, durationMillis, status, exceptionDetail, textLogFilePath, modelLogFilePath, children);
    }

}

File: testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/report_model/ReportStep.kt
package com.testerum.runner.report_model

import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.api.test_context.ExecutionStatus
import java.time.LocalDateTime

data class ReportStep(@JsonProperty("stepCall")         val stepCall: ReportStepCall,
                      @JsonProperty("startTime")        val startTime: LocalDateTime,
                      @JsonProperty("endTime")          val endTime: LocalDateTime,
                      @JsonProperty("durationMillis")   val durationMillis: Long,
                      @JsonProperty("status")           val status: ExecutionStatus,
                      @JsonProperty("textLogFilePath")  override val textLogFilePath: String,
                      @JsonProperty("modelLogFilePath") override val modelLogFilePath: String,
                      @JsonProperty("children")         val children: List<ReportStep>) : RunnerReportNode
